### PR TITLE
operating_mode: fix packet reader function to correctly escape the bytes

### PR DESCRIPTION
**Pull request by hgonzaleDigi**
- Transformed _unescape_data() method to public.
- Removed all unnecesary calls to unescape_data() from create_packet methods.
- Replaced __read_next_packet() method by a search in the xbee packet queue.

Signed-off-by: Héctor González <hector.gonzalez@digi.com>

### CODE REVIEW

**Comment by hgonzaleDigi** [comment]
This pull request adds the fix for issue #15

File: digi/xbee/reader.py [hunk 1]
```diff
@@ -625,18 +625,21 @@ def __execute_user_callbacks(self, xbee_packet, remote=None):
                                                     sender=str(xbee_packet.phone_number),
                                                     more_data=xbee_packet.data))
 
-    def __read_byte_into_packet(self, xbee_packet, operating_mode):
+    def __read_next_byte(self, operating_mode):
         """
-        Reads the next byte and appends it to ``xbee_packet``.
+        Returns the next byte.
```
Comment (diescalo) [inline digi/xbee/reader.py]: I would change this line by something like: 

> Returns the next byte in bytearray format. If the operating mode is  ``OperatingMode.ESCAPED_API_MODE``, the bytearray could contain 2 bytes.

File: digi/xbee/reader.py [hunk 2]
```diff
@@ -625,18 +625,21 @@ def __execute_user_callbacks(self, xbee_packet, remote=None):
...
 
         If in escaped API mode and the byte that was read was the escape byte,
         it will also read the next byte.
```
Comment (diescalo) [inline digi/xbee/reader.py]: Add information about the parameter:
```
Args:
    operating_mode (:class:`.OperatingMode`): the operating mode in which the byte should be read.
```

File: digi/xbee/reader.py [hunk 3]
```diff
@@ -658,14 +671,28 @@ def __try_read_packet(self, operating_mode=OperatingMode.API_MODE):
...
-            # Add packet payload and checksum.
-            for _ in range(0, length + 1):
-                self.__read_byte_into_packet(xbee_packet, operating_mode)
-            return xbee_packet
+            packet_length_byte = bytearray()
+            for _ in range(0, 2):
+                packet_length_byte += self.__read_next_byte(operating_mode)
+            xbee_packet += packet_length_byte
+            # Length needs to be un-escaped to obtain its integer equivalent.
+            length = utils.length_to_int(XBeeAPIPacket.unescape_data(packet_length_byte))
```
Comment (mikewadsten) [inline digi/xbee/reader.py]: This assumes that the LSB of the length is not going to be the escape byte itself (when not in escaped API mode). What happens if the length field is, say, `01 7D`? When traversing that field, `unescape_data` will consume the `01` but ignore `7D`.

**Pull request merged**